### PR TITLE
Allow experiences to be bundled with app

### PR DESCRIPTION
**Pull request by gdamron**
T#1310 Create editor tool to add assets for App Id(s) to Streaming assets folder
T#1378 Load data from Stream Assets folder when available

### CODE REVIEW

File: Assets/Source/Application/BundledAppManifest.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CreateAR.EnkluPlayer
+{
+    [Serializable]
+    public class BundledAppManifest
```
Comment (thegoldenmule) [inline Assets/Source/Application/BundledAppManifest.cs]: Comments.

File: Assets/Source/Application/States/Play/App/AppDataLoader.cs [hunk 2]
```diff
@@ -225,8 +225,8 @@ public void Unload()
             // Setup primary behavior downloads
             _primaryLoadToken = Async.All(
                     LoadPrerequisites(appId, behavior),
-                    LoadScenes(appId, behavior))
-                .OnSuccess(_ => rtnToken.Succeed(Void.Instance))
+                    LoadScenes(appId, behavior));
+            _primaryLoadToken.OnSuccess(_ => rtnToken.Succeed(Void.Instance))
```
Comment (thegoldenmule) [inline Assets/Source/Application/States/Play/App/AppDataLoader.cs]: :D

File: Assets/Source/Application/States/Play/App/AppDataLoader.cs [hunk 3]
```diff
@@ -477,7 +481,7 @@ private void Verbose(string message, params object[] replacements)
         /// </summary>
         /// <param name="data">Asset data.</param>
         /// <returns></returns>
-        private static AssetData ToAssetData(Asset data)
+        public static AssetData ToAssetData(Asset data)
```
Comment (thegoldenmule) [inline Assets/Source/Application/States/Play/App/AppDataLoader.cs]: If these are public, they should be up with the other public methods.

File: Assets/Source/Player/Asset/PassthroughLocalAssetLoader.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+﻿using System.Collections;
+using UnityEngine;
+
+namespace CreateAR.EnkluPlayer
+{
+    /// <summary>
+    /// Passthrough implementation of <c>ILocalAssetLoader</c>
+    /// </summary>
+    class PassthroughLocalAssetLoader : ILocalAssetLoader
```
Comment (thegoldenmule) [inline Assets/Source/Player/Asset/PassthroughLocalAssetLoader.cs]: Missing `public`.

File: Assets/Editor/Tools/EditorWindow/Experiences/AppManifestEditorWindow.cs [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+
+        /// <summary>
+        /// Flags the need to write to disk;
+        /// </summary>
+        private bool _overridden;
+
+        /// <summary>
+        /// Determines whether UI elements should be collapsed.
+        /// </summary>
+        private List<bool> _foldouts = new List<bool>();
```
[the remark this answers is not in the document]
Comment (gdamron) [reply Assets/Editor/Tools/EditorWindow/Experiences/AppManifestEditorWindow.cs]: 👍

File: Assets/Editor/Tools/EditorWindow/Experiences/AppManifestEditorWindow.cs [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+                Filter = APP_MANIFEST_NAME,
+                EnableRaisingEvents = true
+            };
+            _watcher.Changed += Manifest_OnChanged;
+            _watcher.Created += Manifest_OnChanged;
+            _watcher.Deleted += Manifest_OnChanged;
+
+            LoadAppManifest();
+
+            _foldouts = new List<bool>();
```
[the remark this answers is not in the document]
Comment (gdamron) [reply Assets/Editor/Tools/EditorWindow/Experiences/AppManifestEditorWindow.cs]: 👍

File: Assets/Source/Application/BundledAppManifest.cs (hunk 1 above)
Comment (gdamron) [reply Assets/Source/Application/BundledAppManifest.cs]: 👍

File: Assets/Source/Application/States/Play/App/AppDataLoader.cs (hunk 3 above)
Comment (gdamron) [reply Assets/Source/Application/States/Play/App/AppDataLoader.cs]: 👍

**Comment by gdamron** [comment]
@thegoldenmule addressed you comments. I also added a little more logic to ensure credentials are loaded by the editor that could use a look.

**Pull request merged**